src/VisualStudio/Core/Def/ProjectSystem/MetadataReferences/VisualStudioMetadataReferenceManager.cs

```diff
@@ -210,15 +211,16 @@ AssemblyMetadata GetMetadataWorker()
                     StreamCopy(fileStream, copyStream, offset, size);
                 }
 
-                // copy over the data to temp storage and let pooled stream go
-                storage = _temporaryStorageService.CreateTemporaryStreamStorage();
-
+                // Now, copy over the metadata bytes into a memory mapped file.  This will keep it fixed in a single
+                // location, so we can create a metadata value wrapping that.  This will also let us share the memory
+                // for that metadata value with our OOP process.
                 copyStream.Position = 0;
-                storage.WriteStream(copyStream, CancellationToken.None);
+                var handle = _temporaryStorageService.WriteToTemporaryStorage(copyStream, CancellationToken.None);
+                storageIdentifier = handle.Identifier;
             }
 
-            // get stream that owns the underlying unmanaged memory.
-            stream = storage.ReadStream(CancellationToken.None);
+            // Now, read the data from the memory-mapped-file back into a stream that we load into the metadata value.
+            stream = _temporaryStorageService.ReadFromTemporaryStorageService(storageIdentifier, CancellationToken.None);
```

Comment:
this is a pattern you'll see in a few places.  Namely that we dump some object into an MMF, but then we read it right back in.  That's because it's now safe to treat the data in hte MMF as an UmanagedMemoryStream/Pointer since it now will never move.  This is great for this unmanaged data as we don't have to worry about hte lifetime of it from that point on.